cabal-install/src/Distribution/Client/Dependency.hs

```diff
@@ -710,7 +717,8 @@ resolveDependencies platform comp pkgConfigDB solver params =
   $ fmap (validateSolverResult platform comp indGoals)
   $ runSolver solver (SolverConfig reordGoals cntConflicts fineGrained minimize
                       indGoals noReinstalls
-                      shadowing strFlags allowBootLibs onlyConstrained_ maxBkjumps enableBj
+                      shadowing strFlags allowBootLibs nonUpgradeablePackages
```

Comment:
Would it be possible to add some comment as to why the `nonInstallablePackages` field is initialized with `nonUpgradeablePackages` (there’s a clear mismatch here)? Perhaps, reference a place where the difference between noninstallable and nonupgradeable is discussed? @grayjay, do you happen to know such a place? Or if there’s none, perhaps you could help with writing the comment I’m asking for. This stuff is very confusing and I hope we can write up something that could clear things up for the future reference.

Reply:
@ulysses4ever I think that non-installable and non-upgradable may have the exact same meaning, which is the reason I opened #8581. If this PR is merged as is, then I think it can just have a comment pointing to #8581. If it is possible to remove `requireInstalled`, then this part of the change can be reverted.

Reply:
I added a comment

Reply:
If we agree that "non-installable" and "non-upgradable" are the same can we please rename one of these into the the one?

@wismill thanks for the comment. Can you also update the comment on `nonUpgradeablePackages` to say that they are the same?

Reply:
@ulysses4ever I'm not completely sure that non-installable and non-upgradable are the same. I wasn't able to determine the difference by looking at the history of the code, so I think that the best way to answer the question is to try to remove one of them and see whether the tests pass. I think it would be better to just link to issue #8581 until we can try removing the rest of the redundant code.

Reply:
I think the current code fullfils @grayjay comment, so I will wait you reach an agreement 😉.
Once this PR is merged, I can try the proposed [follow-up](https://github.com/haskell/cabal/pull/8712#issuecomment-1416617646).